bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/menus/MenuHelper.java

```diff
@@ -453,62 +373,19 @@ public static MCommand getMCommand(MApplication application, CommandContribution
 		return null;
 	}
 
-	public static String getIconURI(ImageDescriptor descriptor, IEclipseContext context) {
+	public static String getIconURI(ImageDescriptor descriptor, @SuppressWarnings("unused") IEclipseContext context) {
```

Comment:
Why is EclipseContext unused here?

Reply:
See the discussion in #401, where I was asking for feedback how to cope with the fact that with the introduction of Adaters.adapt we no longer need the context. Since I was working before I got feedback, I decided to start with the least invasive approach that keeps the original signature. This might still be useful, if we would consider to optional consider in the future usage of `org.eclipse.e4.core.services.adapter.Adapter.adapt(Object element, Class<T> adapterType)` if a context is provided and Adapters.adapt otherwise. Whether this additional complexity is worth the effort is unclear, because the default implementation of `org.eclipse.e4.core.services.adapter.Adapter.adapt(Object element, Class<T> adapterType)` calls `org.eclipse.core.runtime.Adapters.adapt(Object sourceObject, Class<T> adapter)` anyway.

Reply:
@laeubi I have now made an improvement which gets rid of the unused suppression and uses - if a context is available - now the `org.eclipse.e4.core.services.adapter.Adapter` interface, otherwise as fallback the classical `org.eclipse.core.runtime.Adapters`. Do you consider this as an improvement or would you prefer to go back to the previous proposal (with possibly different adaptions)?